swagger.yaml

```diff
@@ -1,75 +1,164 @@
+          in: "query"
+          description: "ONID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+      responses:
+        200:
+          description: "Permissions successfully retrieved"
+          schema:
+            $ref: "#/definitions/PermissionsResultArray"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
 definitions:
-  Info:
+
+  PermissionLevel:
+    type: "integer"
+    minimum: 0
+    maximum: 5
+    description: "Permission level for a given category"
+    example: 3
+
+  Permissions:
+    type: "object"
+    properties:
+      onid:
+        type: "string"
+        description: "ONID of person who permissions belong to"
+        example: "browni"
+      hrPermissions:
+        $ref: "#/definitions/PermissionLevel"
+      studentPermissions:
+        $ref: "#/definitions/PermissionLevel"
+      financePermissions:
+        $ref: "#/definitions/PermissionLevel"
+
+  SelfLink:
+    type: "object"
+    properties:
+      self:
+        type: "string"
+        format: "url"
+        description: "Self link of resource"
+        example: "https://api.oregonstate.edu/api/v1/permissions/123456789"
+
+  PermissionsResource:
+    type: "object"
+    properties:
+        id:
+          type: "string"
+          description: "OSU ID of person who permissions belong to"
+          example: "123456789"
+        type:
+          type: "string"
+          example: "permissions"
+        attributes:
+          $ref: "#/definitions/Permissions"
+        links:
+          $ref: "#/definitions/SelfLink"
+
+  PermissionsResult:
+    type: "object"
+    properties:
+        data:
+          $ref: "#/definitions/PermissionsResource"
+
+  PermissionsResultArray:
+    type: "object"
     properties:
-      name:
-        type: string
-        description: Name of the API.
-      time:
-        type: string
-        description: Build time in ISO 8601 format.
-      unixTime:
-        type: integer
-        description: Build time as milliseconds since the unix epoch.
-      commit:
-        type: string
-        description: Git commit hash of the revision which was built.
-      documentation:
-        type: string
-        description: Filename of the swagger specification for the API.
+      data:
+        type: "array"
+        items:
+          $ref: "#/definitions/PermissionsResource"
+
   Error:
+    type: "object"
     properties:
       status:
-        type: integer
-        description: HTTP status code.
-      developerMessage:
-        type: string
-        description: An error string aimed at developers.
-      userMesage:
-        type: string
-        description: An error string aimed at end users.
+        type: "integer"
+        description: "HTTP status code"
+        example: 404
```

Comment:
The example value is a little misleading because this object is referenced by other errors, like `500`. Maybe remove the example or change it to something to indicate that it's an integer, like `123`?